### ISSUE TITLE
Fetch Game Cover For Discord Rich Presence

### ISSUE THREAD

**Issue by Overimagine1** [hrydgard/ppsspp#19423]
### What should happen

PPSSPP should fetch the games album cover from a service like [IDGB](https://www.igdb.com/) for it's rich presence. [You are able to use image links for rich presences](https://discord.com/developers/docs/rich-presence/overview#assets).

I have made a quick mockup of what this could look like. I've also made it so it uses the game title instead of "PPSSPP."
![image](https://github.com/user-attachments/assets/733c9a9f-0c6d-440f-bb7a-fed527653a6b)

### Who would this benefit

This is more of a QOL change so the benefit would be small but nice. Discord users who see the PPSSPP rich presence. It is much quicker to recognize the cover art for a game and realize it's the game rather than just the title and the PPSSPP logo. It also just looks a lot nicer. 

### Platform (if relevant)

Other

### Games this would be useful in

All

### Checklist

- [X] Check the [latest git build](https://www.ppsspp.org/devbuilds) in case it's already implemented.
- [X] [Search for other requests](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same feature.

**Comment by hrydgard** [comment]
What are the terms for using images from there? How does it work with copyright? Can we put unlimited load on their servers? (If we can, we could display covers on the games tab...)

Last I checked that site didn't have PSP games, but that was years ago, now it seems it does.

**Comment by anr2me** [comment]
Based on the FAQ, they only interested with data available publicly that can re-distributed, so may be there is no copyright issue?
> Yes. If you have data that we think will complete the overall IGDB data set and you are willing to share that data with us, we can opt for this approach instead. Please be aware, however, that we are only interested in publicly available data that we can re-distribute using this API.

**Comment by Overimagine1** [comment]
> Can we put unlimited load on their servers? (If we can, we could display covers on the games tab...)
[Seems to be 4 requests a second](https://api-docs.igdb.com/#rate-limits). If you're gonna display covers on the game tab it doesn't seem to be an issue if you can download the images needed onto the users PC.

**Comment by anr2me** [comment]
4 request for a single PPSSPP instance might not be an issue, but if it was limited to 4 requests for all PPSSPP users worldwide trying to run PPSSPP at the same time, will certainly cause issue (some of them might get `429 Too Many Requests` error).

However, it's less likely to happen except when a group of players agreed on playing multiplayer at the same time, but that won't happened at the exact same second, and we can always retry a few seconds later (may need to give delay before the next request to prevent naughty players from intentionally floods the server with requests) when a player goes to Games tab again.

**Comment by hrydgard** [comment]
Oh, that's a pretty cool idea. The retrochievements game icon will be cached anyway if you have it enabled and are playing the game, so no additional cost (other than code implementation time...)

**Comment by anr2me** [comment]
Yeah, it looks better, since it can even shows you at which part of the game you're (eg. Main Menu)
